src/js/views/patients/patient/sidebar/sidebar_views.js

```diff
@@ -1,142 +1,136 @@
-      this.showView('item', field.get('value'));
-    });
+  groups: {
+    behaviors: [
+      {
+        behaviorClass: ModelRender,
+        changeAttributes: ['_groups'],
+      },
+    ],
+    template: hbs`{{#each groups}}<div>{{ this.name }}</div>{{/each}}`,
+    templateContext() {
+      return {
+        groups: _.map(this.model.getGroups().models, 'attributes'),
+      };
+    },
   },
-});
-
-const GroupsView = View.extend({
-  behaviors: [
-    {
-      behaviorClass: ModelRender,
-      changeAttributes: ['_groups'],
+  engagement: View.extend({
+    modelEvents: {
+      'change:engagement': 'render',
+    },
+    getTemplate() {
+      if (this.model.get('engagement')) {
+        return hbs`
+          {{#unless notAvailable}}
+            <span class="engagement-status__icon {{ engagement.status }} u-margin--r-4">{{fas "circle"}}</span>{{formatMessage (intlGet "patients.patient.sidebar.sidebarViews.engagementStatusView.status") status=engagement.status}}
+          {{else}}
+            <span class="patient-sidebar__no-engagement">{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusView.notAvailable }}</span>
+          {{/unless}}
+        `;
+      }
+
+      return hbs`<span class="js-loading">{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusPreloadView.loading }}</span>`;
+    },
+    templateContext() {
+      const engagement = this.model.get('engagement');
+      return {
+        notAvailable: engagement && engagement.status === 'Not Available',
+      };
+    },
+    ui: {
+      loading: '.js-loading',
+    },
+    triggers: {
+      'click': 'click',
     },
-  ],
+    onRender() {
+      if (!this.model.get('engagement')) {
+        anime({
+          targets: this.ui.loading[0],
+          opacity: 0.5,
+          loop: true,
+          easing: 'easeInOutSine',
+          duration: 400,
+          direction: 'alternate',
+        });
+      }
+    },
+    onClick() {
+      Radio.request('sidebar', 'start', 'engagement', { patient: this.model });
+    },
+  }),
+};
+
+const widgetView = View.extend({
   className: 'patient-sidebar__section',
-  template: PatientSidebarGroupsTemplate,
-  templateContext() {
-    return {
-      groups: _.map(this.model.getGroups().models, 'attributes'),
-    };
+  template: hbs`{{#if display_name}}<div class="patient-sidebar__heading">{{ display_name }}</div>{{/if}}<div class="patient-sidebar__item" data-content-region></div>`,
+  regions: {
+    content: '[data-content-region]',
   },
-});
-
-const EngagementStatusPreloadView = View.extend({
-  template: hbs`{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusPreloadView.loading }}`,
   onRender() {
-    anime({
-      targets: this.el,
-      opacity: 0.5,
-      loop: true,
-      easing: 'easeInOutSine',
-      duration: 400,
-      direction: 'alternate',
-    });
+    const region = this.getRegion('content');
+    const widget = this.getOption('widget');
+
+    if (_.isFunction(widget)) {
```

Comment:
This is a bit clunky. But since widgets can either be an options object or a view, and also need to have the patient model passed to them, I think we need this check so we know how to append the model.